internal/operators/api/api.go

```diff
@@ -55,6 +56,8 @@ type Operator interface {
 	GetPreflightRequirements(ctx context.Context, cluster *common.Cluster) (*models.OperatorHardwareRequirements, error)
 	// GetFeatureSupportID returns the operator unique feature-support ID
 	GetFeatureSupportID() models.FeatureSupportLevelID
+	// GetBundleLabels returns the bundles associated with the operator
+	GetBundleLabels() pq.StringArray
```

Comment:
I think we should avoid using types from the `pq` library in the contracts: it ties us to that library more than needed. Actually at some point we should migrate from `pq` to `pgx`, as `pq` is no longer maintaned, see https://github.com/lib/pq?tab=readme-ov-file#status . If we want a list of strings then I'd say this should be `[]string`.

It feels a bit strange that in the REST API we represent the relationship of bundles as a list of operator names in the bundle representation, but internally it is a list of labels in operators. Would it be unreasonable to use same representation in the REST API and internally?

Reply:
This will go away as well if we change the bundle to be an object instead of a string.